loader/loader.go

```diff
@@ -307,6 +281,69 @@ func (l *Loader) UnloadWallet() error {
 	return nil
 }
 
+// SetChainClient sets the chain server client.
+func (l *Loader) SetChainClient(chainClient *dcrrpcclient.Client) {
+	l.mu.Lock()
+	l.chainClient = chainClient
+	l.mu.Unlock()
+}
+
+// StartTicketPurchase launches the ticketbuyer to start purchasing tickets.
+func (l *Loader) StartTicketPurchase(passphrase []byte, ticketbuyerCfg *ticketbuyer.Config) error {
+	defer l.mu.Unlock()
+	l.mu.Lock()
+
+	// Already running?
+	if l.purchaseManager != nil {
+		return ErrTicketBuyerStarted
+	}
+
+	if l.wallet == nil {
+		return ErrWalletNotLoaded
+	}
+	w := l.wallet
+	p, err := ticketbuyer.NewTicketPurchaser(ticketbuyerCfg, l.chainClient, w, l.chainParams)
+	if err != nil {
+		return err
+	}
+	if passphrase != nil {
+		var unlockAfter <-chan time.Time
+		err = w.Unlock(passphrase, unlockAfter)
+		if err != nil {
+			return err
+		}
+	}
+	n := w.NtfnServer.MainTipChangedNotifications()
+	pm := ticketbuyer.NewPurchaseManager(w, p, n.C)
+	l.ntfnClient = n
+	l.purchaseManager = pm
+	pm.Start()
+	return nil
+}
+
+// StopTicketPurchase stops the ticket purchaser, waiting until it has finished.
+func (l *Loader) StopTicketPurchase() error {
+	defer l.mu.Unlock()
+	l.mu.Lock()
+
+	if l.purchaseManager != nil {
+		l.ntfnClient.Done()
+		l.purchaseManager.Stop()
+		l.purchaseManager.WaitForShutdown()
+		l.purchaseManager = nil
+		return nil
+	}
+
+	return ErrTicketBuyerStopped
+}
+
+// PurchaseManager returns the ticket purchaser instance.
```

Comment:
it *might* return the ticket purchaser instance.  or it might not if it wasn't set.  comment should at least call out that this returns an optional and the caller must check for nil.